django/urls/resolvers.py

```diff
@@ -322,17 +322,31 @@ def __init__(self, route, name=None, is_endpoint=False):
         self.name = name
 
     def match(self, path):
-        match = self.regex.search(path)
-        if match:
-            # RoutePattern doesn't allow non-named groups so args are ignored.
-            kwargs = match.groupdict()
-            for key, value in kwargs.items():
-                converter = self.converters[key]
-                try:
-                    kwargs[key] = converter.to_python(value)
-                except ValueError:
-                    return None
-            return path[match.end() :], (), kwargs
+        if self.converters:
+            if match := self.regex.search(path):
+                # RoutePattern doesn't allow non-named groups so args are ignored.
+                kwargs = match.groupdict()
+                for key, value in kwargs.items():
+                    converter = self.converters[key]
+                    try:
+                        kwargs[key] = converter.to_python(value)
+                    except ValueError:
+                        return None
+                return path[match.end() :], (), kwargs
+
+        # If there are no converters, skip the regex overhead and
+        # use string operations
+
+        # If this is an endpoint, the path should be
+        # exactly the same as the route
+        elif self._is_endpoint:
+            if self._route == path:
+                return "", (), {}
+
+        # If this isn't an endpoint, the path should start with the route
+        elif path.startswith(self._route):
+            return path.removeprefix(self._route), (), {}
+
```

Comment:
```suggestion
        # If there are no converters, skip the regex overhead and use string operations
        # If this is an endpoint, the path should be exactly the same as the route
        elif self._is_endpoint:
            if self._route == path:
                return "", (), {}
        # If this isn't an endpoint, the path should start with the route
        elif path.startswith(self._route):
            return path.removeprefix(self._route), (), {}
```

Reply:
I disagree - the extra newlines break up each condition, and make it clearer what the comment is for.